File: invision-backend/app/Enums/RouteStatus.php
<?php

namespace App\Enums;

enum RouteStatus: string
{
    case Draft = 'draft';
    case Published = 'published';
    case InProgress = 'in_progress';
    case Completed = 'completed';
    case Cancelled = 'cancelled';

    public function label(): string
    {
        return match ($this) {
            self::Draft => 'Draft',
            self::Published => 'Published',
            self::InProgress => 'In Progress',
            self::Completed => 'Completed',
            self::Cancelled => 'Cancelled',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Draft => 'gray',
            self::Published => 'blue',
            self::InProgress => 'yellow',
            self::Completed => 'green',
            self::Cancelled => 'red',
        };
    }
}

File: invision-mobile/lib/src/core/enums/route_status.dart
enum RouteStatus {
  draft,
  published,
  inProgress,
  completed,
  cancelled;

  String get label => switch (this) {
    draft => 'Draft',
    published => 'Published',
    inProgress => 'In Progress',
    completed => 'Completed',
    cancelled => 'Cancelled',
  };

  static RouteStatus fromString(String value) {
    return switch (value) {
      'draft' => RouteStatus.draft,
      'published' => RouteStatus.published,
      'in_progress' => RouteStatus.inProgress,
      'completed' => RouteStatus.completed,
      'cancelled' => RouteStatus.cancelled,
      _ => RouteStatus.draft,
    };
  }
}
